controllers/azurevirtualmachine_controller.go

```diff
@@ -17,7 +19,7 @@ type AzureVirtualMachineReconciler struct {
 // +kubebuilder:rbac:groups=azure.microsoft.com,resources=azurevirtualmachines,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=azure.microsoft.com,resources={azurevirtualmachines/status,azurevirtualmachines/finalizers},verbs=get;update;patch
 
-func (r *AzureVirtualMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+func (r *AzureVirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
Do you not get lint errors about `ctx` being unused on all of these?

Reply:
I don't think we have such a linter for ASO v1. Although I am pretty sure we don't have one for ASO v2 either? You have a pretty good point that I should pass it down though...

Reply:
I confirmed that we do not have a linter that catches this currently enabled for the generator or generated (V2 ASO).
I have also passed this ctx to the function called here, which means it's now plumbed all the way through (previously the function called here manufactured a context from `context.Background()`)